server/sandbox_run_linux.go

```diff
@@ -552,70 +570,56 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
-		}
-	}()
-
-	if s.defaultIDMappings != nil && !s.defaultIDMappings.Empty() {
-		rootPair := s.defaultIDMappings.RootPair()
-		for _, path := range pathsToChown {
-			if err := os.Chown(path, rootPair.UID, rootPair.GID); err != nil {
-				return nil, errors.Wrapf(err, "cannot chown %s to %d:%d", path, rootPair.UID, rootPair.GID)
+	if !container.Spoofed() {
+		s.addInfraContainer(container)
+		defer func() {
+			if err != nil {
+				s.removeInfraContainer(container)
 			}
-		}
-	}
+		}()
 
-	if err := s.createContainerPlatform(container, sb.CgroupParent()); err != nil {
-		return nil, err
-	}
+		if err := s.createContainerPlatform(container, sb.CgroupParent()); err != nil {
+			return nil, err
+		}
 
-	if err := s.Runtime().StartContainer(container); err != nil {
-		return nil, err
-	}
+		if err := s.Runtime().StartContainer(container); err != nil {
+			return nil, err
+		}
 
-	defer func() {
-		if err != nil {
-			// Clean-up steps from RemovePodSanbox
-			timeout := int64(10)
-			if err2 := s.Runtime().StopContainer(ctx, container, timeout); err2 != nil {
-				log.Warnf(ctx, "failed to stop container %s: %v", container.Name(), err2)
-			}
-			if err2 := s.Runtime().WaitContainerStateStopped(ctx, container); err2 != nil {
-				log.Warnf(ctx, "failed to get container 'stopped' status %s in pod sandbox %s: %v", container.Name(), sb.ID(), err2)
-			}
-			if err2 := s.Runtime().DeleteContainer(container); err2 != nil {
-				log.Warnf(ctx, "failed to delete container %s in pod sandbox %s: %v", container.Name(), sb.ID(), err2)
-			}
-			if err2 := s.ContainerStateToDisk(container); err2 != nil {
-				log.Warnf(ctx, "failed to write container state %s in pod sandbox %s: %v", container.Name(), sb.ID(), err2)
+		defer func() {
+			if err != nil {
+				// Clean-up steps from RemovePodSanbox
+				timeout := int64(10)
+				if err2 := s.Runtime().StopContainer(ctx, container, timeout); err2 != nil {
+					log.Warnf(ctx, "failed to stop container %s: %v", container.Name(), err2)
+				}
+				if err2 := s.Runtime().WaitContainerStateStopped(ctx, container); err2 != nil {
+					log.Warnf(ctx, "failed to get container 'stopped' status %s in pod sandbox %s: %v", container.Name(), sb.ID(), err2)
+				}
+				if err2 := s.Runtime().DeleteContainer(container); err2 != nil {
+					log.Warnf(ctx, "failed to delete container %s in pod sandbox %s: %v", container.Name(), sb.ID(), err2)
+				}
+				if err2 := s.ContainerStateToDisk(container); err2 != nil {
+					log.Warnf(ctx, "failed to write container state %s in pod sandbox %s: %v", container.Name(), sb.ID(), err2)
+				}
 			}
+		}()
+
+		if err := s.ContainerStateToDisk(container); err != nil {
+			log.Warnf(ctx, "unable to write containers %s state to disk: %v", container.ID(), err)
 		}
-	}()
 
-	if err := s.ContainerStateToDisk(container); err != nil {
-		log.Warnf(ctx, "unable to write containers %s state to disk: %v", container.ID(), err)
+		if err := s.MonitorConmon(container); err != nil {
```

Comment:
I noticed the MonitorConmon() call originally on line 638 has been removed.
Do we still want the call here ?

Reply:
no we've dropped conmonmon